### PR TITLE
Add numexpr 3

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Add a new pacakge numexpr3 since it's not available by specifying a version of the old py-numexpr pacakge. (Hopefully Cmake support can be added later! )

### CODE REVIEW

**Comment by s-sajid-ali** [comment]
How do I set preferred version to 3.0.1a1 ?

File: var/spack/repos/builtin/packages/py-numexpr3/package.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    use less memory than doing the same calculation in Python.
+    In addition, its multi-threaded capabilities can make use of all your
+    cores, which may accelerate computations, most specially if they are not
+    memory-bounded (e.g. those using transcendental functions).
+    Compared to NumExpr 2.6, functions have been re-written in a fashion such
+    that gcc can auto-vectorize them with SIMD instruction sets such as
+    SSE2 or AVX2, if your processor supports them. Use of a newer version of
+    gcc such as 5.4 is strongly recommended."""
+    homepage = "https://github.com/pydata/numexpr/tree/numexpr-3.0"
+    url = "https://files.pythonhosted.org/packages/f2/fe/52b6aca524622e57fb1fa9681c6e12bf3dd7df42c93edc2900ae844d42ce/numexpr3-3.0.1a1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numexpr3/package.py]: Can you switch this URL to https://pypi.io/packages/source/n/numexpr3/numexpr3-3.0.1a1.tar.gz? Then Spack can automatically detect and replace the version correctly.

File: var/spack/repos/builtin/packages/py-numexpr3/package.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    version('3.0.1.a1', '9fa8dc59b149aa1956fc755f982a78ad')
+    # TODO: Add CMake build system for better control of passing flags related
+    # to CPU ISA.
+
+    depends_on('python@2.6:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numexpr3/package.py]: I think this should actually be:
```python
depends_on('python@2.6:2.8,3.3:', type=('build', 'run'))
```

**Comment by s-sajid-ali** [comment]
For some reason, if I do 
```
spack install py-numexpr3 
```
there's some error, but this works : 
```
spack install py-numexpr3@3.0.1a1
```
I wonder if changing the url might fixex that.

**Comment by adamjstewart** [comment]
What error?

**Comment by s-sajid-ali** [comment]
I think I made the mistake of switching the branch before doing that. Apologies for the error.

**Pull request merged**